PixivConfig.py

```diff
@@ -11,648 +12,192 @@
+        ConfigItem("Ugoira", "createGif", False),
+        ConfigItem("Ugoira", "createApng", False),
+        ConfigItem("Ugoira", "deleteUgoira", False),
+        ConfigItem("Ugoira", "createWebm", False),
+        ConfigItem("Ugoira", "createWebp", False),
+
+        ConfigItem("DownloadControl", "minFileSize", 0),
+        ConfigItem("DownloadControl", "maxFileSize", 0),
+        ConfigItem("DownloadControl", "overwrite", False),
+        ConfigItem("DownloadControl", "backupOldFile", False),
+        ConfigItem("DownloadControl", "dayLastUpdated", 7),
+        ConfigItem("DownloadControl", "alwaysCheckFileSize", False),
+        ConfigItem("DownloadControl", "checkUpdatedLimit", 0),
+        ConfigItem("DownloadControl", "useBlacklistTags", False),
+        ConfigItem("DownloadControl", "dateDiff", 0),
+        ConfigItem("DownloadControl", "enableInfiniteLoop", False),
+        ConfigItem("DownloadControl", "useBlacklistMembers", False),
+        ConfigItem("DownloadControl", "downloadResized", False),
+    ]
+
+    proxy = {"http": "", "https": "", }
+
+    def __init__(self):
+        for item in self.__items:
+            setattr(self, item.option, item.default)
 
     def loadConfig(self, path=None):
-        ''' New settings must be added on the last part.'''
-
         if path is not None:
             self.configFileLocation = path
         else:
             self.configFileLocation = script_path + os.sep + 'config.ini'
 
         print('Reading', self.configFileLocation, '...')
-        haveError = False
         config = configparser.RawConfigParser()
-        try:
-            config.read_file(PixivHelper.open_text_file(self.configFileLocation))
-
-            self.username = config.get('Authentication', 'username')
-            self.password = config.get('Authentication', 'password')
-            self.cookie = config.get('Authentication', 'cookie')
-
-            self.tagsSeparator = config.get('Filename', 'tagsseparator')
-            self.rootDirectory = os.path.expanduser(config.get('Settings', 'rootdirectory'))
-
-            try:
-                self.IrfanViewPath = os.path.expanduser(config.get('IrfanView', 'IrfanViewPath'))
-                self.downloadListDirectory = os.path.expanduser(config.get('Settings', 'downloadListDirectory'))
-            except BaseException:
-                pass
-
-            try:
-                self.processFromDb = config.getboolean('Settings', 'processfromdb')
-            except ValueError:
-                print("processFromDb = True")
-                self.processFromDb = True
-                haveError = True
-
-            try:
-                self.proxyAddress = config.get('Network', 'proxyaddress')
-            except ValueError:
-                print("proxyAddress = ''")
-                self.proxyAddress = ''
-                haveError = True
-            self.proxy = {'http': self.proxyAddress, 'https': self.proxyAddress}
```

Comment:
missed out this line for followup call?

Reply:
Hmm.... you mean `self.proxy` should be a followup of `proxyAddress`?

Reply:
ah, see it in: https://github.com/Nandaka/PixivUtil2/blob/b97865208fdf9cacc6855d2186ece453a715cc23/PixivConfig.py#L196

Reply:
I think you can keep the current implementation to keep it simple first, else you need to modify the followup lambda to pass the caller, so it can set the attributes in the caller.

Reply:
Actually this could be done with `followup` as well and it should be better.
Nice point. Haha

Reply:
okay... but I think I maybe need to set `self.proxy` in `__init__` as well, do you think so?